controller/GmBinToBin.controller.js

```diff
@@ -0,0 +1,298 @@
+				path: "Werks",
+				operator: sap.ui.model.FilterOperator.EQ,
+				value1: werks
+			});
+			var l_filter3 = new sap.ui.model.Filter({
+				path: "Lgpla",
+				operator: sap.ui.model.FilterOperator.EQ,
+				value1: lgpla
+			});
+			larray_filters.push(l_filter1);
+			larray_filters.push(l_filter2);
+			larray_filters.push(l_filter3);
+			var filterStr = "$filter=Lgnum eq " + lgnum + " and Werks eq " + werks + " and Lgpla eq " + lgpla;
+			var oDataModel = this.getView().getModel("GiUnp");
+			oDataModel.read("/BinMaterialsSet", {
+				filters: larray_filters,
+				success: function(oData) {
+					var oDataResultModel = new sap.ui.model.json.JSONModel();
+					oDataResultModel.setData(oData);
+					that.getView().setModel(oDataResultModel, "matTable");
+					if (!that._oDialog) {
+						that._oDialog = sap.ui.xmlfragment("smud.org.wmLM01.fragments.material", that);
+					}
+					that.getView().addDependent(that._oDialog);
+					jQuery.sap.syncStyleClass("sapUiSizeCompact", that.getView(), that._oDialog);
+					sap.ui.core.BusyIndicator.hide();
+					that._oDialog.open();
+				},
+				error: function(oError) {
+					//MessageBox.show(JSON.parse(oError.responseText).error.innererror.errordetails[0].message,{ title: "Error Message"});
+					MessageBox.show(JSON.parse(oError.responseText), {
+						title: "Error",
+						icon: "ERROR"
+					});
+				}
+
+			});
+
+			sap.ui.core.BusyIndicator.hide();
+
+		},
+		handleMaterialSelect: function(oEvent) {
+			var oModel = this.getView().getModel();
+			var oBinding = oEvent.getSource().getBinding("items");
+			oBinding.filter([]);
+			var aContexts = oEvent.getParameter("selectedContexts");
+			if (aContexts && aContexts.length) {
+				var selectedMaterialNbr = aContexts.map(function(oContext) {
+					return oContext.getObject().Matnr;
+				});
+				var selectedSourceType = aContexts.map(function(oContext) {
+					return oContext.getObject().Lgtyp;
+				});
+				oModel.setProperty("/Matnr", selectedMaterialNbr);
+				oModel.setProperty("/Lgtyp", selectedSourceType[0].toString());
+				oModel.setProperty("/Vltyp", selectedSourceType[0].toString());
+
+				zController.gotoBinToBin();
+			}
+		},
+		handleMaterialSearch: function(oEvent) {
+			var searchStr = oEvent.getParameter("value");
+			zController.filterTable(oEvent.getSource(), ["Matnr", "Maktg", "Meins"], searchStr);
+		},
+		filterTableSearch: function(oTable, fields, filterString) {
+			// filter list
+			var oBinding = oTable.getBinding("items");
+			if (oBinding) {
+				var oFilters = [];
+				for (var i = 0; i < fields.length; i++) {
+					oFilters.push(new sap.ui.model.Filter(fields[i], sap.ui.model.FilterOperator.Contains, filterString));
+				}
+				var filterObj = new sap.ui.model.Filter(oFilters, false);
+				oBinding.filter(filterObj);
+			} else {
+				oBinding.filter([]);
+			}
+		},
+		uppercase: function(oEvent) {
+			GlobalFuncs.uppercase(oEvent);
+		},
+		goBack: function() {
+			window.history.go(-1);

```

Comment:
use a BaseController for the implementations of the goBack method... that way you can reuse it in all your controllers